### DISCUSSION TITLE
Trouble with parked car count sensor & frigate /recordings/

### DISCUSSION THREAD

**Discussion by spacebares** [blakeblackshear/frigate#4592] [Ask A Question] [answered]
I have 2 vehicles that are inside a zone labeled parking (pink). As you can see it is not active because "car" is not tracked by the zone. Frigate picks both vehicles up regardless which is great because now when they cross the yellow "driveway" zone, it never misses to generate event no matter how fast they go or how bad the % is due to rain etc.

![image](https://user-images.githubusercontent.com/57186372/205483669-e7d3833a-1e35-4178-ad80-14062b3fc20b.png)

There was a problem with this setup where the event would stay in progress forever. I solved it with `stationary -> max_frames -> car -> 900`.

There is 2 problems now i have discovered. Since both vehicles are being tracked, home assistant puts car count to 2. Garage lights work off this sensor and will turn on... Not so much an issue normally as they turn on then turn off after a delay even if the sensor is still reading 2 cars, but frigate is not perfect. Frigate will lose sight of both vehicles for a split second and then reacquire them. **This happens every 900 frames.** The result which causes the lights to never turn off: 
![image](https://user-images.githubusercontent.com/57186372/205483403-64f4fcb3-e23f-412b-9ff1-b4b154c3edbe.png)
If i take out the stationary max_frames there is a solid 2 count now for the vehicles but now the event will forever be in-progress. This in-progress event would have hours of footage too. This is what that looks like:

![image](https://user-images.githubusercontent.com/57186372/205484387-54143735-9f6d-4d70-9a26-c0c4e70c74fc.png)

And for recordings in general, even though no new event is being generated and the old parked car events that cross driveway -> parking is not receiving any new footage.. frigate is still recording tons of 4K footage of these parked vehicles filling up the drive. There's 4 hours of just bugs, squirrels and who knows what but nothing of interest as you can imagine since no event has been generated. Is this an issue for long-term storage? One vehicle can linger there for 1-2 days and then it leaves, While the other vehicle comes and goes multiple times a day. I want to keep footage for events for as long as i can but i don't recorded footage of nothing just piling up for 100 days.

These are the record settings:
```
record:
  enabled: True
  retain:
    days: 100
    mode: motion
  events:
    pre_capture: 5
    post_capture: 30
```

**Comment by NickM-27** [comment]
> Frigate will lose sight of both vehicles for a split second and then reacquire them. This happens every 900 frames. The result which causes the lights to never turn off:

This is exactly what you're telling frigate to do, and why that setting is explained as only being meant for false positives and not for actual objects. 

> If i take out the stationary max_frames there is a solid 2 count now for the vehicles but now the event will forever be in-progress. This in-progress event would have hours of footage too.

If you don't want this then you should be using a different record retention mode https://docs.frigate.video/configuration/record#what-do-the-different-retain-modes-mean

**Reply by spacebares** [reply]
Going to tryout these settings and see what happens:

```
record:
  enabled: True
  retain:
    days: 1
#  mode: all
    mode: motion
  events:
    pre_capture: 5
    post_capture: 30
    retain:
      default: 365
      mode: motion
```
edit: motion on both works great

**Reply by spacebares** [reply]
Updated both to motion only, saved 200gb of space. Old events still viewable. Really long events seem to disappear or shrink(?) but all is well now.

**Comment by blakeblackshear** [comment]
What you are seeing is exactly why stationary object tracking was implemented. It seems to be better to keep frigate aware of existing objects rather than have it rediscover them repeatedly.

I would consider adjusting your automation to use the events topic and fire when objects make the transition from the driveway to the parking zone rather than using the count of objects in the parking zone.

**Reply by spacebares** [reply]
So the event is just supposed to stay in-progress forever? Even though car crossed from driveway into parking? I can't use required zones because then persons in driveway are missed.

**Reply by NickM-27** [reply]
Yes, an event lasts as long as the object is visible. 

> I can't use required zones because then persons in driveway are missed.

You can set different objects for different zones, I don't think that should be an issue.

**Reply by spacebares** [reply]
I removed the stationary max_frames from config, and of course now the event stays in progress until the vehicle leaves. However tonight i got 3 notifications in a 5 hour span for the same parked vehicle in a zone that does not track vehicles (debug shows no thickened line). I will be enabling max_frames again since it both clears the in-progress event and does not notify me again until someone is near the vehicle or when it crosses driveway again. I will increase max_frames for about 10 minutes worth and should be enough to keep vehicles tracked as they cross driveway since they are always re-acquired when a person comes in frame.

I will look into a different automation approach like you suggested for the lights. I believe [this](https://docs.frigate.video/guides/ha_notifications#conditions) is what you are talking about.

A tuned Frigate runs so good that i can leave phone notifications on at night. I know it runs good because all the false motion & critters skittering about at night that is captured by the Camera's OEM NVR is ignored by frigate. And frigate still able to capture cars & people at night without buzzing the phone because they are not in "notify" zones. Great work is being done on this project ❤️.

**Comment by spacebares** [comment, marked as answer]
ye

**Reply by spacebares** [reply]
Both answers are correct above but i choose ye.
